File: Day3/Star1.py
if __name__ == '__main__':
    '''
    data = ['00100', '11110', '10110', '10111', '10101',
            '01111', '00111', '11100', '10000', '11001',
            '00010', '01010'];
    '''
    file = open('data.txt')
    data = file.readlines()
    file.close()


    gamma = [0] * (len(data[0])-1)
    epsilon = [0] * (len(data[0])-1)
    for line in data:
        line = line.strip()
        for i in range(0,len(line)-1):
            gamma[i] += int(line[i])
    gamma = [1 if i>(len(data)/2) else 0 for i in gamma]
    epsilon = [0 if i else 1 for i in gamma]
    gamma_res = 0
    epsilon_res = 0
    for i in range(0, len(data[0])-1):
        gamma_res += gamma[-i-1]*2**i;
        epsilon_res += epsilon[-i-1]*2**i;

    print(gamma_res)
    print(epsilon_res)
    print(gamma_res*epsilon_res)

File: Day03/Star1.kt
import java.io.File
import java.io.InputStream

fun main() {
    val inputStream: InputStream = File("data.txt").inputStream();
    val lineList = mutableListOf<String>();
    inputStream.bufferedReader().forEachLine { lineList.add(it) }
    var gamma = IntArray(lineList.first().length);
    var epsilon = IntArray(lineList.first().length);
    lineList.forEach{
      for(i in it.indices){
          val value : Int = (it[i].code - '0'.code);
          gamma[i] += value;
          epsilon[i] += if(value==1) 0 else 1;
      }
    }
    gamma = transform(gamma, lineList.size/2);
    val gamma_res : Int = toInt(gamma);
    epsilon = transform(epsilon, lineList.size/2);
    val epsilon_res : Int = toInt(epsilon);

    println(gamma_res);
    println(epsilon_res);
    println(gamma_res*epsilon_res);
}

fun transform(value: IntArray, comp: Int) : IntArray {
    for (i in value.indices){
      value[i] = if(value[i] > comp) 1 else 0;
    }
    return value;
}

fun toInt(value: IntArray) : Int {
    var res : Int = 0;
    val reversed_value : IntArray = value.reversedArray();
    for (i in reversed_value.indices){
      res += reversed_value[i]*(Math.pow(2.toDouble(), i.toDouble()).toInt());
    }
    return res;
}
